app/recipe/tests/test_ingredients_api.py

```diff
@@ -0,0 +1,70 @@
+from django.urls import reverse
+from django.test import TestCase
+
+from rest_framework import status
+from rest_framework.test import APIClient
+
+from core.models import Ingredient, Recipe
+
+from recipe.serializers import IngredientSerializer
+
+
+INGREDIENTS_URL = reverse('recipe:ingredient-list')
+
+
+class IngredientsApiTests(TestCase):
+    """Tests ingredients api"""
+
+    def setUp(self):
+        self.client = APIClient()
+
+    def test_retrieve_ingredient_list(self):
+        """Test retrieving a list of the ingredients"""
+        Ingredient.objects.create(name='Kale')
+        Ingredient.objects.create(name='Salt')
+
+        res = self.client.get(INGREDIENTS_URL)
+
+        ingredients = Ingredient.objects.all()
+        serializer = IngredientSerializer(ingredients, many=True)
+
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertEqual(res.data, serializer.data)
+
+    def test_create_ingredient_successful(self):
+        """Test create a new ingredient"""
+        payload = {'name': 'Cabbage'}
+        self.client.post(INGREDIENTS_URL, payload)
+
+        exists = Ingredient.objects.filter(
```

Comment:
Ideally you should test that it does not exist before `post`, just to exclude the case, that it was in some kind of setUp method (not the case here, but in production code you never know :wink: )

You can reuse the queryset, e.g.
```
qs = Ingredient.objects.filter(name="Cabbage")
self.assertFalse(qs.exists())
# do the post
self.assertTrue(qs.exists())
```

Reply:
hahhaa yes, you are right, will add it right away! :D